### DISCUSSION TITLE
Getting weight of a MLP Model

### DISCUSSION THREAD

**Discussion by atale480** [pyg-team/pytorch_geometric#6916] [Q&A] [answered]
How can I get weight of a MLP model from torch.geometric.nn ?

**Comment by rusty1s** [comment, marked as answer]
What do you mean by weights? The weights of the `Linear` layers? You can get them via
```python
weights = [lin.weight for lin in mlp.lins]
```

**Reply by atale480** [reply]
As it is built-in class, I thought I could get weight of whole MLP model in one tensor.

**Reply by rusty1s** [reply]
That's nothing we thought about yet. Not totally sure I see the use-case yet, but if it would help we could integrate this with your help :)
